lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -15,6 +15,10 @@ include_once 'dosomething_user.theme.inc';
  */
 function dosomething_user_preprocess_page(&$vars) {
   $school_api_endpoint = variable_get('dosomething_user_school_api_endpoint', 'http://lofischools.herokuapp.com/search');
+  if ($_SESSION['log_login']) {
```

Comment:
I'm not sure exactly what `log_login` tracks -- any comments we can add here? -- but the basic rule is that, when external cache is present, un-sessioned (unauthenticated) requests will almost never reach the back end.